src/test/queue.spec.ts

```diff
@@ -0,0 +1,138 @@
+import * as sinon from "sinon";
+import * as chai from "chai";
+
+chai.use(require("chai-as-promised"));
+const { expect } = chai;
+
+const Queue = require("../queue");
+
+const TEST_ERROR = new Error("foobar");
+
+describe("Queue", () => {
+  it("should ignore non-number backoff", () => {
+    const q = new Queue({
+      backoff: "not a number"
+    });
+    expect(q.backoff).to.equal(200);
+  });
+
+  it("should handle function tasks", () => {
+    const task = sinon.stub().resolves();
+    const q = new Queue({});
+
+    q.add(task);
+    q.close();
+
+    return q.wait()
+        .then(() => {
+          expect(task.callCount).to.equal(1);
+          expect(q.complete).to.equal(1);
+          expect(q.success).to.equal(1);
+          expect(q.errored).to.equal(0);
+          expect(q.retried).to.equal(0);
+        });
+  });
+
+  it("should handle tasks", () => {
+    const handler = sinon.stub().resolves();
+    const q = new Queue({
+      handler,
+    });
+
+    q.add(4);
+    q.close();
+
+    return q.wait()
+      .then(() => {
+        expect(handler.callCount).to.equal(1);
+        expect(q.complete).to.equal(1);
+        expect(q.success).to.equal(1);
+        expect(q.errored).to.equal(0);
+        expect(q.retried).to.equal(0);
+      });
+  });
+
+  it("should not retry", () => {
+    const handler = sinon.stub().rejects(TEST_ERROR);
+    const q = new Queue({
+      handler,
+      retries: 0,
+    });
+
+    q.add(4);
+    q.close();
+
+    return q.wait()
+      .then(() => {
+        throw new Error("handler should have rejected");
+      })
+      .catch((err: Error) => {
+        expect(err).to.equal(TEST_ERROR);
+      })
+      .then(() => {
+        expect(handler.callCount).to.equal(1);
+        expect(q.complete).to.equal(1);
+        expect(q.success).to.equal(0);
+        expect(q.errored).to.equal(1);
```

Comment:
BTW, I don't think we can ever have `errored > 1`, because node has no concurrency issue.
It always call `_finish()` when an error is found.

So I would recommend removing this stats.

Reply:
@bkendall Thoughts on this?

Reply:
I don't quite understand your question: you want to remove `errored` as a property? I think we should keep it, even if it doesn't ever go over `1` as it's a useful indication of failure.